test/demo/open.inttest.ts

```diff
@@ -84,21 +81,98 @@ describe('Opening applications with the FDC3 client', () => {
-                await expect(fin.Application.wrapSync(testAppWithPreregisteredListeners1).isRunning()).resolves.toBe(true);
+                    // Check that the app is now running
+                    await expect(fin.Application.wrapSync(testAppWithPreregisteredListeners1).isRunning()).resolves.toBe(true);
 
-                const preregisteredListener = await fdc3Remote.getRemoteContextListener(testAppWithPreregisteredListeners1);
+                    const preregisteredListener = await fdc3Remote.getRemoteContextListener(testAppWithPreregisteredListeners1);
 
-                // Check that the app received the context passed in open and nothing else
-                await expect(preregisteredListener).toHaveReceivedContexts([validContext]);
+                    // Check that the app received the context passed in open and nothing else
+                    await expect(preregisteredListener).toHaveReceivedContexts([validContext]);
+                });
+
+                test('When the app adds its listener after a short delay, the app opens and its context listener is triggered with the \
+correct data', async () => {
+                    // From the launcher app, call fdc3.open with a valid name and context
+                    const openPromise = open(testAppInDirectory1.name, validContext);
+
+                    // Wait a short delay after the app is running
+                    await waitForAppToBeRunning(testAppInDirectory1);
+                    await delay(2500);
+
+                    // Add a listener
+                    const listener = await fdc3Remote.addContextListener(testAppInDirectory1);
+
+                    await openPromise;
+
+                    // Check that the app received the context passed in open and nothing else
+                    await expect(listener).toHaveReceivedContexts([validContext]);
+                });
+
+                test('When the app adds its listener on a child window, the app opens and its context listener is triggered with the \
+correct data', async () => {
+                    // From the launcher app, call fdc3.open with a valid name and context
+                    const openPromise = open(testAppInDirectory1.name, validContext);
+
+                    await waitForAppToBeRunning(testAppInDirectory1);
+
+                    // Create a child window and add a listener on it
+                    const childWindow = await fdc3Remote.createFinWindow(testAppInDirectory1, {url: testAppUrl, name: 'child-window'});
+                    const listener = await fdc3Remote.addContextListener(childWindow);
+
+                    await openPromise;
+
+                    // Check that the app received the context passed in open and nothing else
+                    await expect(listener).toHaveReceivedContexts([validContext]);
+                });
+
+                test('When the app adds listeners on mulitple windows, the app opens and the first windows\'s context listener is \
```

Comment:
`window's?` Also how come punctuation here but no where else

Reply:
`mulitple` as well

Reply:
Spelling corrected. Not sure what punctuation you mean though

Reply:
✅